### PR TITLE
test: DataTablesPane

### PR DESCRIPTION
**Pull request by yardz**
### SUMMARY
Tests for DataTablesPane component

### TEST PLAN
No behavior changes. All tests must pass.

### CODE REVIEW

File: superset-frontend/src/explore/components/DataTablesPane/DataTablesPane.test.tsx [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import userEvent from '@testing-library/user-event';
+import React from 'react';
+import { render, screen, act } from 'spec/helpers/testing-library';
+import { DataTablesPane } from '.';
+
+let defaultProps = {
```
Comment (michael-s-molina) [inline superset-frontend/src/explore/components/DataTablesPane/DataTablesPane.test.tsx:25]: If the props are not modified, you can change it to a `const` and remove `beforeEach`. If only some are being modified, you can override them inside each test.

File: superset-frontend/src/explore/components/DataTablesPane/DataTablesPane.test.tsx [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+  expect(screen.getByRole('img', { name: 'right' })).toBeVisible();
+});
+
+test('Shoud show tabs', async () => {
+  render(<DataTablesPane {...defaultProps} />, { useRedux: true });
+  expect(screen.queryByText('View results')).not.toBeInTheDocument();
+  expect(screen.queryByText('View samples')).not.toBeInTheDocument();
+  userEvent.click(await screen.findByText('Data'));
+  expect(screen.queryByText('View results')).toBeVisible();
+  expect(screen.queryByText('View samples')).toBeVisible();
```
Comment (michael-s-molina) [inline superset-frontend/src/explore/components/DataTablesPane/DataTablesPane.test.tsx:113]:
```suggestion
  expect(screen.getByText('View results')).toBeVisible();
  expect(screen.getByText('View samples')).toBeVisible();
```

File: superset-frontend/src/explore/components/DataTablesPane/DataTablesPane.test.tsx [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+});
+
+test('Shoud show tabs: View results', async () => {
+  render(<DataTablesPane {...defaultProps} />, {
+    useRedux: true,
+  });
+  userEvent.click(await screen.findByText('Data'));
+  userEvent.click(await screen.findByText('View results'));
+  expect(screen.queryByText('0 rows retrieved')).toBeVisible();
```
Comment (michael-s-molina) [inline superset-frontend/src/explore/components/DataTablesPane/DataTablesPane.test.tsx:122]:
```suggestion
  expect(screen.getByText('0 rows retrieved')).toBeVisible();
```

File: superset-frontend/src/explore/components/DataTablesPane/DataTablesPane.test.tsx [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+test('Shoud show tabs: View samples', async () => {
+  render(<DataTablesPane {...defaultProps} />, {
+    useRedux: true,
+  });
+  userEvent.click(await screen.findByText('Data'));
+  expect(screen.queryByText('0 rows retrieved')).not.toBeInTheDocument();
+  await act(async () => {
+    userEvent.click(await screen.findByText('View samples'));
+  });
+  expect(screen.queryByText('0 rows retrieved')).toBeVisible();
```
Comment (michael-s-molina) [inline superset-frontend/src/explore/components/DataTablesPane/DataTablesPane.test.tsx:134]:
```suggestion
  expect(screen.getByText('0 rows retrieved')).toBeVisible();
```

File: superset-frontend/src/explore/components/DataTablesPane/index.tsx [hunk 5]
```diff
@@ -255,7 +255,7 @@ export const DataTablesPane = ({
   };
 
   return (
-    <SouthPane>
+    <SouthPane data-test="DataTablesPane">
```
Comment (geido) [inline superset-frontend/src/explore/components/DataTablesPane/index.tsx:258]: Super small nit. I think I have seen a pretty adopted convention for the `data-test` to be hyphen separated or just lower case

**Review by geido — COMMENTED** [review]
LGTM except for a nit and Michael's comments

File: superset-frontend/src/explore/components/DataTablesPane/DataTablesPane.test.tsx (hunk 1 above)
Comment (yardz) [reply superset-frontend/src/explore/components/DataTablesPane/DataTablesPane.test.tsx:25]: Same reason here: https://github.com/apache/superset/pull/13753#discussion_r600729737

File: superset-frontend/src/explore/components/DataTablesPane/index.tsx (hunk 5 above)
Comment (yardz) [reply superset-frontend/src/explore/components/DataTablesPane/index.tsx:258]: I agree with you, but I chose to go that way for the same reason here:  https://github.com/apache/superset/pull/13638#discussion_r600734118

File: superset-frontend/src/explore/components/DataTablesPane/DataTablesPane.test.tsx (hunk 1 above)
Comment (yardz) [reply superset-frontend/src/explore/components/DataTablesPane/DataTablesPane.test.tsx]: done: https://github.com/apache/superset/pull/13753#discussion_r600862661

File: superset-frontend/src/explore/components/DataTablesPane/index.tsx (hunk 5 above)
Comment (rusackas) [reply superset-frontend/src/explore/components/DataTablesPane/index.tsx:258]: Ahh, this is a little counter to my comment/thought on that thread. In the linked example, the data-test attr uses the compnent name _inside_ the component (at the outer DOM layer), so any instance of the component will have that data-test attr. I'm fine with that.

In this case, this is a _particular, purposeful instance_ of _a_ SouthPane component. In this case, I think we should follow the convention and go with the `data-test="some-purposeful-instance"` convention.

Comment (yardz) [reply superset-frontend/src/explore/components/DataTablesPane/index.tsx]: Okay, I don't have such a strong opinion. As they are already "2 against" we will keep it that way :-)

**Review by rusackas — APPROVED** [review]
LGTM

**Pull request merged**